### PR TITLE
issue 1550: changed behaviour of "hzn util configconv"

### PR DESCRIPTION
**Pull request by mikelo**
added default path horizon/hzn.json for "hzn util configconv" command and removed mandatory clause for parameter "-f"

I also committed a change to the go.mod file that got automatically modified during the build process, hope that's ok

please review my changes, thanks!

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -33,3 +33,4 @@ go.sum
 /anax-in-container/LICENSE.txt
 /agent-install/test/config/agent-install.cfg                                                                                                                               /agent-install/test/config/agent-install.crt                                                                                                                               /agent-install/test/config/switch/agent-install.cfg                                                                                                                        /agent-install/test/config/switch/agent-install.crt
 /agent-install/test/config/switch/agent-install-test-results.log
+cli/horizon/
```
Comment (linggao) [inline .gitignore:36]: why is this added?

File: cli/hzn.go [hunk 2]
```diff
@@ -585,7 +587,7 @@ Environment Variables:
 	utilVerifyPubKeyFile := utilVerifyCmd.Flag("public-key-file", msgPrinter.Sprintf("The path of public key file (that corresponds to the private key that was used to sign) to verify the signature of stdin.")).Short('K').Required().ExistingFile()
 	utilVerifySig := utilVerifyCmd.Flag("signature", msgPrinter.Sprintf("The supposed signature of stdin.")).Short('s').Required().String()
 	utilConfigConvCmd := utilCmd.Command("configconv", msgPrinter.Sprintf("Convert the configuration file from JSON format to a shell script."))
-	utilConfigConvFile := utilConfigConvCmd.Flag("config-file", msgPrinter.Sprintf("The path of a configuration file to be converted. ")).Short('f').Required().ExistingFile()
+	utilConfigConvFile := utilConfigConvCmd.Flag("config-file", msgPrinter.Sprintf("The path of a configuration file to be converted. If not specified horizon/hzn.json will be used")).Default("horizon/hzn.json").Short('f').ExistingFile()
```
Comment (linggao) [inline cli/hzn.go:590]: horizon/hzn.json does not always exists.  Please do not default it to `horizon/hzn.json`. And change the text to "This flag can be omitted if it is running from a project directory created by 'hzn dev' command. In this case, the project configuration file hzn.json will be used."

**Review by linggao — CHANGES_REQUESTED** [review]
per my comments.

File: cli/hzn.go (hunk 2 above)
Comment (linggao) [reply cli/hzn.go:590]: The logic for finding the hzn.json file is:  if the command is run under the project_dir,  project_dir/horizon, or project_dir/horizon/dependencies directories, the hzn.json for this project is used. See `hzn dev -h` for more help.

**Comment by linggao** [comment]
@mikelo could you fix this PR?

File: .gitignore (hunk 1 above)
Comment (mikelo) [reply .gitignore:36]: sorry this is a mistake, please ignore this

File: cli/hzn.go (hunk 2 above)
Comment (mikelo) [reply cli/hzn.go:590]: > horizon/hzn.json does not always exists. Please do not default it to `horizon/hzn.json`. And change the text to "This flag can be omitted if it is running from a project directory created by 'hzn dev' command. In this case, the project configuration file hzn.json will be used."
> For implementation, in ConvertConfig() function of cli/utilcmds/utilcmds.go, first you need to get the project config file if cofigFile is empty. You can call cliconfig.GetProjectConfigFile("") with an empty string as the input. if it returns an empty string, please ask the use to use -f, other wise use the returned string as the config file.

this change does not require `horizon/hzn.json`, I will definitely use the text you provided.

Comment (mikelo) [reply cli/hzn.go:590]: > The logic for finding the hzn.json file is: if the command is run under the project_dir, project_dir/horizon, or project_dir/horizon/dependencies directories, the hzn.json for this project is used. See `hzn dev -h` for more help.

this logic seems to differ from the original issue description, I could definitely add this logic to the one in the original issue.

Comment (linggao) [reply cli/hzn.go:590]: 1. Please do not use `Default("horizon/hzn.json")` here. This will allow the file name to be empty when no specified by -f flag. 
2. In the https://github.com/open-horizon/anax/blob/master/cli/utilcmds/utilcmds.go#L41, if `cofigFile` is not empty, keep the old logic. However, if `cofigFile` is empty, call `cliconfig.GetProjectConfigFile("")` (https://github.com/open-horizon/anax/blob/master/cli/cliconfig/config.go#L282) to get the project config file name. If the project config file name is empty, tell user that they need to specify the file using -f and exit out. If the project config file name is not empty, use it as the input, i.e. `cofigFile` to proceed.
3. As to how `cliconfig.GetProjectConfigFile("")` gets the project config file name, you need to understand the structure of a project. You can create a directory, say myservice,  and run the following command under this new directory. 
````hzn dev service new -o myorg -s myservice -i myrepo/myimage```` 
Then you can see the directory structure of a project. The `cliconfig.GetProjectConfigFile("")` function gets the hzn.json when you run `hzn util configconv` under the following directories:

- myservice
- myservice/horizon
- myservice/horizon/dependencies

Comment (linggao) [reply cli/hzn.go:590]: @mikelo please see my latest comments.
